content-copy-tool/contentcopytool/lib/operation_objects.py

```diff
@@ -203,6 +203,17 @@ def __init__(self, server, credentials):
         self.server = server
         self.credentials = credentials
 
+    def credentials_valid(self):
```

Comment:
I think you can rework this function and the one in role_updates.py so the code doing the actual check is not duplicated. Put this code in a separate function and call it from this one and the one in role_updates.py

>print("Testing credentials: %s - %s" % (self.credentials, self.server))
        response = http.http_get_request(auth_req_url, auth=auth)
        if response.status_code == 401:
            print("Bad credentials: %s - %s" % (self.credentials, self.server))
            return False
        print("Good credentials.")
        return True

Reply:
I had the same thought. But I had no idea where to put such a common function. Any ideas as to where it might go instead? content_copy.py, util.py?

Reply:
util.py would be fine.